codersdk/agentconn.go

```diff
@@ -171,8 +175,11 @@ type ReconnectingPTYInit struct {
 	Command string
 }
 
-func (c *AgentConn) ReconnectingPTY(id string, height, width uint16, command string) (net.Conn, error) {
-	conn, err := c.DialContextTCP(context.Background(), netip.AddrPortFrom(TailnetIP, uint16(TailnetReconnectingPTYPort)))
+func (c *AgentConn) ReconnectingPTY(ctx context.Context, id string, height, width uint16, command string) (net.Conn, error) {
+	ctx, span := tracing.StartSpan(ctx)
+	defer span.End()
+
+	conn, err := c.DialContextTCP(ctx, netip.AddrPortFrom(TailnetIP, uint16(TailnetReconnectingPTYPort)))
```

Comment:
Might be nice to capture the errors in the span too.

Reply:
There are a lot of errors across the codebase we would have to catch, so it's kinda weird to only do it here. It'd be nice if we could hook into all returned error values in Golang but I don't think that's possible without writing and enforcing a wrapper for errors instead of xerrors.Errorf